sdk/src/handshake.rs

```diff
@@ -0,0 +1,53 @@
+use std::cell::Cell;
+
+use ic_kit::{
+    interfaces::{management, Method},
+    Principal,
+};
+
+use crate::CapEnv;
+
+/// Handshakes with Cap. This either grabs important metadata or
+/// creates the root bucket for this contract and gives it `creation_cycles`
+/// cycles.
+pub fn handshake(creation_cycles: u64) {
+    let arg = management::CreateCanisterArgument { settings: None };
+
+    thread_local! {
+        static CYCLES: Cell<u64> = Cell::new(0);
+    }
+
+    CYCLES.with(|cycles| cycles.set(creation_cycles));
```

Comment:
Because the lifetime of the u64 has to be static, and that's the only way to convince it that it is as far as I can see. Why does it have to be static? Because the lifetime of the futures container is static. Why does *that* have to be static? Because I tried to put it in the env but the futures container isn't clone which is required for storing it.